std/algorithm/searching.d

```diff
@@ -3462,6 +3462,67 @@ unittest
 }
 
 /**
+Computes the index of the first occurrence of `range`'s minimum (respectively maximum).
+
+Params:
+    pred = The ordering predicate to use to determine the extremum (minimum or
+           maximum) element.
+    range = The input range to search.
+
+Returns:
+    The index of the first encounter of the minimum (respectively maximum) in `range`. If the
+    range is empty, -1 is returned.
+ */
+sizediff_t minIndex(alias pred = "a < b", Range)(Range range)
+    if (isForwardRange!Range && !isInfinite!Range &&
+        is(typeof(binaryFun!pred(range.front, range.front))))
+{
+    if (range.empty) return -1;
+
+    sizediff_t minPos = 0, curPos = 0;
+    Unqual!(typeof(range.front)) min = range.front;
+    for (range.popFront(); !range.empty; range.popFront())
+    {
+        ++curPos;
+        if (binaryFun!pred(range.front, min))
+        {
+            min = range.front;
+            minPos = curPos;
+        }
+    }
+    return minPos;
+}
+
+///Ditto
+sizediff_t maxIndex(alias pred = "a < b", Range)(Range range)
+    if (isInputRange!Range && !isInfinite!Range &&
+        is(typeof(binaryFun!pred(range.front, range.front))))
+{
+    return range.minIndex!((a, b) => binaryFun!pred(b, a));
+}
+
+///
+@safe unittest
+{
+    int[] a = [2, 3, 4, 1, 2, 4, 1, 1, 2];
+    int[] b;
+    const int[] c = [2, 5, 4, 1, 2, 3];
+
+    // Minimum is 1 and first occurs in position 3
+    assert(a.minIndex == 3);
+    // Maximum is 4 and first occurs in position 2
+    assert(a.maxIndex == 2);
+    // Range is empty, so return value is -1
+    assert(b.minIndex == -1);
```

Comment:
maybe it's just me, but I do prefer splitting this into separate `unittest` blocks - they then will also generated more nice on dlang.org under the "Examples" section & it also helps the reader to keep track of the arrays.
At the very least it would be nice to  group them like here:

```d
int[] a = [2, 3, 4, 1, 2, 4, 1, 1, 2];

// Minimum is 1 and first occurs in position 3
 assert(a.minIndex == 3);

 int b[];
 assert(b.minIndex == -1);
```